defold-rive/commonsrc/vertices.cpp

```diff
@@ -70,164 +155,11 @@ void GetRiveDrawParams(rive::HContext ctx, rive::HRenderer renderer, uint32_t& v
     }
 }
 
-void SetStencilDrawState(dmRender::StencilTestParams* params, bool is_clipping, bool clear_clipping_flag)
-{
-    params->m_Front = {
-        .m_Func     = dmGraphics::COMPARE_FUNC_ALWAYS,
-        .m_OpSFail  = dmGraphics::STENCIL_OP_KEEP,
-        .m_OpDPFail = dmGraphics::STENCIL_OP_KEEP,
-        .m_OpDPPass = dmGraphics::STENCIL_OP_INCR_WRAP,
-    };
-
-    params->m_Back = {
-        .m_Func     = dmGraphics::COMPARE_FUNC_ALWAYS,
-        .m_OpSFail  = dmGraphics::STENCIL_OP_KEEP,
-        .m_OpDPFail = dmGraphics::STENCIL_OP_KEEP,
-        .m_OpDPPass = dmGraphics::STENCIL_OP_DECR_WRAP,
-    };
-
-    params->m_Ref                = 0x00;
-    params->m_RefMask            = 0xFF;
-    params->m_BufferMask         = 0xFF;
-    params->m_ColorBufferMask    = 0x00;
-    params->m_ClearBuffer        = 0;
-    params->m_SeparateFaceStates = 1;
-
-    if (is_clipping)
-    {
-        params->m_Front.m_Func = dmGraphics::COMPARE_FUNC_EQUAL;
-        params->m_Back.m_Func  = dmGraphics::COMPARE_FUNC_EQUAL;
-        params->m_Ref          = 0x80;
-        params->m_RefMask      = 0x80;
-        params->m_BufferMask   = 0x7F;
-    }
-
-    if (clear_clipping_flag)
-    {
-        params->m_ClearBuffer = 1;
-    }
-}
-
-
-void GetBlendFactorsFromBlendMode(dmRiveDDF::RiveModelDesc::BlendMode blend_mode, dmGraphics::BlendFactor* src, dmGraphics::BlendFactor* dst)
-{
-    switch (blend_mode)
-    {
-        case dmRiveDDF::RiveModelDesc::BLEND_MODE_ALPHA:
-            *src = dmGraphics::BLEND_FACTOR_ONE;
-            *dst = dmGraphics::BLEND_FACTOR_ONE_MINUS_SRC_ALPHA;
-        break;
-
-        case dmRiveDDF::RiveModelDesc::BLEND_MODE_ADD:
-            *src = dmGraphics::BLEND_FACTOR_ONE;
-            *dst = dmGraphics::BLEND_FACTOR_ONE;
-        break;
-
-        case dmRiveDDF::RiveModelDesc::BLEND_MODE_MULT:
-            *src = dmGraphics::BLEND_FACTOR_DST_COLOR;
-            *dst = dmGraphics::BLEND_FACTOR_ONE_MINUS_SRC_ALPHA;
-        break;
-
-        case dmRiveDDF::RiveModelDesc::BLEND_MODE_SCREEN:
-            *src = dmGraphics::BLEND_FACTOR_ONE_MINUS_DST_COLOR;
-            *dst = dmGraphics::BLEND_FACTOR_ONE;
-        break;
-
-        default:
-            dmLogError("Unknown blend mode: %d\n", blend_mode);
-            assert(0);
-        break;
-    }
-}
-
-static void SetRenderObject_DrawStencil(
-    const rive::PathDrawEvent&  evt,
-    bool                        clear_clipping_flag,
-    uint32_t                    ix_start, // bytes
-    uint32_t                    ix_count,
-    // out parameters
-    // RiveVertex*                 vx_ptr,
-    // int*                        ix_ptr,
-    dmRender::RenderObject*     ro)
+uint32_t ProcessRiveEvents(rive::HContext ctx, rive::HRenderer renderer, RiveVertex* vx_ptr, int* ix_ptr,
```

Comment:
We now have a single loop that outputs render calls for either the runtime or the editor.
This loop handles the state-flipping, as well as the vertex/index copying